torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -103,7 +103,7 @@ void DistAutogradContext::addOutstandingRpc(
         if (futErr) {
           // If we have an error, let the local autograd engine know about it.
           std::runtime_error err((*futErr).what());
-          graphTask_->set_exception(std::make_exception_ptr(err), nullptr);
+          graphTask_->set_exception(err, nullptr);
```

Comment:
Hmm? What's going on here?

Reply:
The generic Future uses `setError` to mark exceptions and there is no way to retrieve an error from exception_ptr. As a result, I ensured we instead always pass a reference to the exception.